src/garage/_dtypes.py

```diff
@@ -230,9 +233,25 @@ def __new__(cls, env_spec, observations, last_observations, actions,
                 'step_types tensor must be dtype `StepType`, but got tensor '
                 'of dtype {} instead.'.format(step_types.dtype))
 
+        # episode_infos
+        for key, val in episode_infos.items():
+            if not isinstance(val, (dict, np.ndarray)):
+                raise ValueError(
+                    'Each entry in episode_infos must be a numpy array or '
+                    'dictionary, but got key {} with value type {} instead.'
+                    'instead'.format(key, type(val)))
+
+            if (isinstance(val, np.ndarray)
+                    and val.shape[0] != inferred_batch_size):
```

Comment:
this is incorrect -- inferred_batch_size is the number of time steps, but we need "number of episodes" episode_infos instead (i.e. `num_episodes = len(lengths)`)